eZ/Publish/Core/Repository/Helper/RelationProcessor.php

```diff
@@ -70,12 +71,16 @@ public function appendFieldRelations(
 
                 if (isset($destinationIds['locationIds'])) {
                     foreach ($destinationIds['locationIds'] as $locationId) {
-                        if (!isset($locationIdToContentIdMapping[$locationId])) {
-                            $location = $this->persistenceHandler->locationHandler()->load($locationId);
-                            $locationIdToContentIdMapping[$locationId] = $location->contentId;
-                        }
+                        try {
+                            if (!isset($locationIdToContentIdMapping[$locationId])) {
+                                $location = $this->persistenceHandler->locationHandler()->load($locationId);
+                                $locationIdToContentIdMapping[$locationId] = $location->contentId;
+                            }
 
-                        $relations[$relationType][$locationIdToContentIdMapping[$locationId]] = true;
+                            $relations[$relationType][$locationIdToContentIdMapping[$locationId]] = true;
+                        } catch (NotFoundException $e) {
+                            // Ignore missing relation
```

Comment:
The side effect of this change is that relations with missing target location will be silently removed.

Reply:
Let's inject logger `@?logger` (use NullLogger by default) and log this case.